src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultLSPEditorService.cs

```diff
@@ -89,8 +110,13 @@ public DefaultLSPEditorService(JoinableTaskContext joinableTaskContext, SVsServi
 
                     if (windowFrame != null)
                     {
-                        var textView = GetActiveVsTextView(windowFrame);
-                        MoveCaretToPosition(textView, cursorPosition);
+                        var vsTextView = GetActiveVsTextView(windowFrame);
+
+                        // Since we are moving the cursor we should dismiss any existing completion sessions as that is no longer valid.
+                        var textView = _adaptersFactoryService.GetWpfTextView(vsTextView);
+                        _completionBroker.DismissAllSessions(textView);
```

Comment:
What sort of completion items were being shown? Makes me think one of our completion item providers are providing completions when they shouldn't be.

Reply:
No that isn't what's happening. The completion dialog that opens up when someone types `@` is still open even after we apply the onTypeFormatting edits and move the cursor. But we don't want that to be open.

Reply:
ooo interesting. Is that a bug in the LSP client code?

Reply:
Possibly?